secure/storage/src/value.rs

```diff
@@ -2,27 +2,36 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use crate::error::Error;
-use libra_crypto::ed25519::Ed25519PrivateKey;
+use libra_crypto::{ed25519::Ed25519PrivateKey, hash::HashValue};
 use serde::{Deserialize, Serialize};
 
 #[derive(Debug, Deserialize, PartialEq, Serialize)]
 #[serde(content = "value", rename_all = "snake_case", tag = "type")]
 pub enum Value {
     Ed25519PrivateKey(Ed25519PrivateKey),
+    HashValue(HashValue),
     U64(u64),
 }
 
 impl Value {
-    pub fn u64(self) -> Result<u64, Error> {
-        if let Value::U64(value) = self {
+    pub fn ed25519_private_key(self) -> Result<Ed25519PrivateKey, Error> {
+        if let Value::Ed25519PrivateKey(value) = self {
```

Comment:
nit: I think it might be cleaner and more rusty for all three methods (ed25519_private_key, hash_value, u64) to use a match statement internally: e.g.,

match self {
    Value::HashValue(value) => Ok(value),
    _ => Err(Error::UnexpectedValueType),
}

Reply:
Hummm.... why is that more rusty :P